google-cloud-contrib/google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageReadChannel.java

```diff
@@ -135,9 +135,33 @@ public int read(ByteBuffer dst) throws IOException {
     }
   }
 
+  private static boolean isReopenable(Throwable exs) {
+    Throwable throwable = exs;
+    while (throwable != null) {
+      if (throwable.getMessage().contains("Connection closed prematurely")
+          || throwable.getCause() instanceof javax.net.ssl.SSLException
+          || throwable.getCause() instanceof java.io.EOFException
+          || throwable.getCause() instanceof java.net.SocketException
+          || throwable.getCause() instanceof java.net.SocketTimeoutException) {
+        return true;
+      }
+      throwable = throwable.getCause();
+    }
+    return false;
+  }
+
   private void sleepForAttempt(int attempt) {
+    long delay = 1000;
+    long maxDelay = 120_000;
+    // exponential backoff
+    for (int i=0; i<attempt; i++) {
+      delay *= 2;
```

Comment:
Not sure that 2 as an exponential backoff factor is good - it is too aggressive. Across google-cloud-java we usually use backoff factor of 1.2 or 1.3.

Also, it is probably better to calculate delay on each iteration, to avoid this loop on each wait. For example, the essential changes would be: 
- switch to `private void sleepForAttempt(double delay)`; 
- `delay = Math.min(delay * 1.3, maxDelay);` 
- `Thread.sleep((long) delay)`